pkg/cli/plugins_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("resolvePluginsByKey", func() {
+
+	plugins := makePluginsForKeys(
+		"foo.example.com/v1.0",
+		"bar.example.com/v1.0",
+		"baz.example.com/v1.0",
+		"foo.kubebuilder.io/v1.0",
+		"foo.kubebuilder.io/v2.0",
+		"bar.kubebuilder.io/v1.0",
+		"bar.kubebuilder.io/v2.0",
+	)
+
+	It("should resolve keys without error", func() {
+		By("Resolving foo.example.com/v1.0")
+		resolvedPlugins, err := resolvePluginsByKey(plugins, "foo.example.com/v1.0")
+		Expect(err).NotTo(HaveOccurred())
+		Expect(makePluginKeySlice(resolvedPlugins...)).To(Equal([]string{"foo.example.com/v1.0"}))
+
+		By("Resolving foo.example.com")
+		resolvedPlugins, err = resolvePluginsByKey(plugins, "foo.example.com")
+		Expect(err).NotTo(HaveOccurred())
+		Expect(makePluginKeySlice(resolvedPlugins...)).To(Equal([]string{"foo.example.com/v1.0"}))
+
+		By("Resolving baz")
+		resolvedPlugins, err = resolvePluginsByKey(plugins, "baz")
+		Expect(err).NotTo(HaveOccurred())
+		Expect(makePluginKeySlice(resolvedPlugins...)).To(Equal([]string{"baz.example.com/v1.0"}))
+
+		By("Resolving foo/v2.0")
+		resolvedPlugins, err = resolvePluginsByKey(plugins, "foo/v2.0")
+		Expect(err).NotTo(HaveOccurred())
+		Expect(makePluginKeySlice(resolvedPlugins...)).To(Equal([]string{"foo.kubebuilder.io/v2.0"}))
+	})
+
+	It("should resolve keys with errors", func() {
+
```

Comment:
IMO the tests should be :  

`should works successfully when <scenario>`
`should fail when <scenario>`

Since it makes clear what is expected and when.  could we shape these testes?